feat(stats): add C implementation for `stats/base/dists/chi/variance`

anandkaranubc opened the pull request:
### Resolves  
https://github.com/stdlib-js/stdlib/issues/3499  

---

### Description  

#### Purpose of this pull request  
This pull request:  
- Adds a C implementation of the chi distribution's variance.  
- Includes benchmarks for the C implementation.  
- Provides examples demonstrating the usage of the chi distribution's variance in C.  

---

### Related Issues  

#### Does this pull request have any related issues?  
This pull request:  
- Resolves [Issue #3499](https://github.com/stdlib-js/stdlib/issues/3499).  

---

### Questions  

#### Any questions for reviewers of this pull request?  
No.  

---

### Other Information  

#### Any other relevant information?  
Requires #4514 to be merged first to implement `stats/base/dists/chi/variance`

---

### Checklist  

Please ensure the following tasks are completed before submitting this pull request:  
- [x] Read, understood, and followed the [contributing guidelines](https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md).  

---  
**@stdlib-js/reviewers**

anandkaranubc committed:
feat: add the C src files

anandkaranubc committed:
feat: add the C benchmark and example files

anandkaranubc committed:
feat: add the JS native files

anandkaranubc committed:
build: added the json and gyp dependency files

anandkaranubc committed:
docs: update README with C API documentation

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

Neerajpathak07 commented on lib/node_modules/@stdlib/stats/base/dists/chi/variance/manifest.json:
```diff
@@ -0,0 +1,79 @@
+      "dependencies": [
+        "@stdlib/math/base/napi/unary",
+        "@stdlib/math/base/assert/is-nan",
+        "@stdlib/stats/base/dists/chi/mean"
```
@anandkaranubc 
I think you don't really need to add `@stdlib/stats/base/dists/chi/mean` to the the manifest file which is why it is throwing an error.

anandkaranubc replied:
Hii @Neerajpathak07,

Thanks for the comment. Although, I do see it added over there.

The reason I kept this in draft mode is because I am waiting on https://github.com/stdlib-js/stdlib/pull/4514 to be merged first, which is the main reason why the tests are failing.

Neerajpathak07 commented on lib/node_modules/@stdlib/stats/base/dists/chi/variance/src/main.c:
```diff
@@ -0,0 +1,50 @@
+
+#include "stdlib/math/base/assert/is_nan.h"
+#include "stdlib/stats/base/dists/chi/mean.h"
+
```
@anandkaranubc 
Oh I see you have missed #include `stdlib/stats/base/dists/chi/variance.h` which is why I thought this was the mean implementation.

Neerajpathak07 replied:
The test cases are indeed failing as there is no C implementation for `stdlib/stats/base/dists/chi/mean`. Which you were working on so yeah looks good now.

Neerajpathak07 replied:
as without adding ` #include stdlib/stats/base/dists/chi/variance.h` you won't be able to reference `stdlib_base_dists_chi_variance`

Planeshifter commented:
@anandkaranubc C implementation for `@stdlib/stats/base/dists/chi/mean` is now available!

anandkaranubc committed:
Merge branch 'develop' into rfc/chi-variance

anandkaranubc commented:
/stdlib update-copyright-years

stdlib-bot committed:
chore: update copyright years

anandkaranubc committed:
chore: follow code conventions

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: passed
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: passed
  - task: lint_c_benchmarks
    status: passed
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc committed:
style: fix indentation

Signed-off-by: Karan Anand <119553199+anandkaranubc@users.noreply.github.com>

anandkaranubc committed:
chore: fix header file

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc committed:
Update native.js

Signed-off-by: Karan Anand <119553199+anandkaranubc@users.noreply.github.com>

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/chi/variance/README.md:
```diff
@@ -116,6 +116,98 @@ for ( i = 0; i < 10; i++ ) {
+
+The function accepts the following arguments:
+
+-   **k**: `[in] double` degrees of freedom
```
```suggestion
-   **k**: `[in] double` degrees of freedom.
```

Planeshifter reviewed (dismissed):
LGTM.

Planeshifter committed:
docs: add missing period

Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>

Planeshifter reviewed (approved):
LGTM.
